provider/rediscloud_essentials_database_test.go

```diff
@@ -0,0 +1,182 @@
+					resource.TestCheckResourceAttrSet(resourceName, "redis_version_compliance"),
+					resource.TestCheckResourceAttr(resourceName, "resp_version", "resp2"),
+					resource.TestCheckResourceAttr(resourceName, "data_persistence", "none"),
+					resource.TestCheckResourceAttr(resourceName, "data_eviction", "volatile-lru"),
+					resource.TestCheckResourceAttr(resourceName, "replication", "false"),
+					resource.TestCheckResourceAttrSet(resourceName, "public_endpoint"),
+					resource.TestCheckResourceAttr(resourceName, "private_endpoint", ""),
+
+					// Can't get this to work
+					resource.TestCheckResourceAttr(resourceName, "source_ips.#", "0"),
+					// resource.TestCheckResourceAttr(resourceName, "source_ips.0", "192.168.12.0/24"),
+
+					resource.TestCheckResourceAttr(resourceName, "alert.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "alert.0.name", "throughput-higher-than"),
+					resource.TestCheckResourceAttr(resourceName, "alert.0.value", "80"),
+
+					// Even though the user has specified this as an empty list, the API does its own thing here.
+					// resource.TestCheckResourceAttr(resourceName, "modules.#", "0"),
+
+					resource.TestCheckResourceAttr(resourceName, "enable_default_user", "true"),
+					resource.TestCheckResourceAttr(resourceName, "password", "j43589rhe39f"),
+
+					// Test the datasource
+					resource.TestMatchResourceAttr(datasourceName, "id", regexp.MustCompile("^\\d+/\\d+$")),
+					resource.TestCheckResourceAttrSet(datasourceName, "subscription_id"),
+					resource.TestCheckResourceAttrSet(datasourceName, "db_id"),
+					resource.TestCheckResourceAttr(datasourceName, "name", databaseName),
+					resource.TestCheckResourceAttr(datasourceName, "protocol", "stack"),
+					resource.TestCheckResourceAttr(datasourceName, "cloud_provider", "AWS"),
+					resource.TestCheckResourceAttr(datasourceName, "region", "eu-west-1"),
+					resource.TestCheckResourceAttrSet(datasourceName, "redis_version_compliance"),
+					resource.TestCheckResourceAttr(datasourceName, "resp_version", "resp2"),
+					resource.TestCheckResourceAttr(datasourceName, "data_persistence", "none"),
+					resource.TestCheckResourceAttr(datasourceName, "data_eviction", "volatile-lru"),
+					resource.TestCheckResourceAttr(datasourceName, "replication", "false"),
+					resource.TestCheckResourceAttrSet(datasourceName, "public_endpoint"),
+					resource.TestCheckResourceAttr(datasourceName, "private_endpoint", ""),
+					resource.TestCheckResourceAttr(datasourceName, "source_ips.#", "0"),
+					resource.TestCheckResourceAttr(datasourceName, "alert.#", "1"),
+					resource.TestCheckResourceAttr(datasourceName, "alert.0.name", "throughput-higher-than"),
+					resource.TestCheckResourceAttr(datasourceName, "alert.0.value", "80"),
+					// Even though the user has specified this as an empty list, the API does its own thing here.
+					// resource.TestCheckResourceAttr(datasourceName, "modules.#", "0"),
```

Comment:
should this be commented out? maybe it can be removed if the test is not meaningful for this resource